python/lsst/pipe/analysis/colorAnalysis.py

```diff
@@ -363,281 +564,1057 @@ def plotGalaxyColors(self, catalogs, filenamer, dataId):
+                          " < " + str(self.config.analysis.magThreshold) + "]", ha="right", color="blue",
+                          **kwargs)
+                yLoc -= 0.05*deltaY
+                axes.text(xLoc, yLoc, "NinPerpGood =", ha="left", color="red", **kwargs)
+                axes.text(xLoc + fdx*deltaX, yLoc, str(len(xColor[inPerpGood])) + " [" + self.fluxFilter +
+                          " < " + str(self.config.analysis.magThreshold) + "]", ha="right", color="red",
+                          **kwargs)
+                if camera is not None:
+                    labelCamera(camera, plt, axes, 0.5, 1.09)
+                if hscRun is not None:
+                    axes.set_title("HSC stack run: " + hscRun, color="#800080")
+
+                tractStr = "tract: {:d}".format(dataId["tract"])
+                axes.annotate(tractStr, xy=(0.5, 1.04), xycoords="axes fraction", ha="center", va="center",
+                              fontsize=10, color="green")
+
+                fig.savefig(filename, dpi=120)
+                plt.close(fig)
+
+    def plotStarColorColor(self, principalColCats, byFilterCats, filenamer, dataId, fluxColumn, butler=None,
+                           camera=None, tractInfo=None, patchList=None, hscRun=None):
+        num = len(list(byFilterCats.values())[0])
         zp = 0.0
-        mags = {ff: zp - 2.5*np.log10(catalogs[ff][self.fluxColumn]) for ff in catalogs}
+        mags = {ff: zp - 2.5*np.log10(byFilterCats[ff][fluxColumn]) for ff in byFilterCats}
 
         bad = np.zeros(num, dtype=bool)
-        for cat in catalogs.values():
+        for cat in byFilterCats.values():
             for flag in self.flags:
                 if flag in cat.schema:
                     bad |= cat[flag]
 
-        bright = np.ones(num, dtype=bool)
-        for mm in mags.values():
-            bright &= mm < self.config.analysis.magThreshold
+        bright = mags[self.fluxFilter] < self.config.analysis.magThreshold
+        prettyBrightThreshold = self.config.analysis.magThreshold
+        prettyBright = mags[self.fluxFilter] < prettyBrightThreshold
 
+        # Determine number of filters object is classified as a star
         numStarFlags = np.zeros(num)
-        for cat in catalogs.values():
+        for cat in byFilterCats.values():
             numStarFlags += np.where(cat[self.classificationColumn] < 0.5, 1, 0)
 
-        good = (numStarFlags == len(catalogs)) & np.logical_not(bad) & bright
-
-        combined = self.transformCatalogs(catalogs, straightTransforms, hscRun=hscRun)[good].copy(True)
-        filters = set(catalogs.keys())
-        color = lambda c1, c2: (mags[c1] - mags[c2])[good]
+        # Select as a star if classified as such in self.config.fluxFilter
+        isStarFlag = np.where(byFilterCats[self.fluxFilter][self.classificationColumn] < 0.5, True, False)
+        # Require stellar classification in self.fluxFilter and at least one other filter for fits
```

Comment:
Personal pref, if `bad` is a boolean array, then `np.logical_not(bad)` == `~bad`

Reply:
This one relies on using bitwise operators to manipulate booleans, which works but is rather sleazy.

Reply:
Ok, so what is preferred:
``` foo = ~bar1 & (bar2 > 1)```
[and thus can do ```foo = ~bar1 & (bar2 > 1) & bar3 & ...```]
or
```foo = np.logical_and(~bar1, bar2 > 1)```
[and thus would have to also do ```foo = np.logical_and(foo, bar3)```, etc.]
?
i.e. The "sleaziness" of the former allows for multiple comparisons...

Reply:
My favorite is `np.logical_and.reduce([no.logical_not(bar1), bar2 > 1, bar3 < 2])`, which also allows for multiple comparisons while being careful about bitwise vs. logical operations.

Reply:
Ah, so you prefer `np.logical_not(bar1)` to `~bar1` (is that a general preference and thus different to Yusra's above)?  I guess it would seem weird to mix & match...

Reply:
I do prefer `np.logical_not` (but I was mostly making sure you knew about `.reduce`, which makes using `logical_and` and `logical_or` much less onerous).

Reply:
I tried using the .reduce and sticking to Jim's consistent use of np.logical_..., but I was getting type errors I couldn't figure out, so I reverted to Yusra's preference.  If this turns out to be in violation of standards, I can do a repo-wide fix on the 'make ready to move to lsst' ticket.